features/cellular/framework/targets/QUECTEL/BG96/QUECTEL_BG96_CellularContext.cpp

```diff
@@ -70,7 +70,7 @@ ControlPlane_netif *QUECTEL_BG96_CellularContext::get_cp_netif()
 nsapi_error_t QUECTEL_BG96_CellularContext::do_user_authentication()
 {
     uint8_t type = (uint8_t)_pdp_type;
-    if ((uint8_t)_pdp_type < 1) {
+    if (((uint8_t)_pdp_type < 1) || ((uint8_t)_pdp_type > 2)) {
```

Comment:
According to BG96_TCP/IP_AT_Commands_Manual_V1.1 context_type IPv4v6 (value 3) is also valid for `+QICSGP`.

Reply:
This value 3 didn't work with my modem... Maybe this v1.1 needs some specific modem version ?

Reply:
Yes, BG96 firmware update could help?